### PR TITLE
Making mergifyLabelPaths overridable

### LINKED ISSUE #220: Can't disable mergify auto-labeling
Setting `ThisBuild / mergifyLabelPaths := Map.empty` doesn't disable labeling.

### PR DESCRIPTION
**Pull request by TonioGela**
Closes #220.

### CODE REVIEW

**Comment by TonioGela** [comment]
Hmm, running `mergifyGenerate` annihilates `.mergify.yml`, so the pr logic is probably wrong 👉 👈

File: mergify/src/main/scala/org/typelevel/sbt/mergify/MergifyPlugin.scala [hunk 1]
```diff
@@ -114,8 +115,8 @@ object MergifyPlugin extends AutoPlugin {
       .dependsOn(ThisBuild / mergifyGenerate)
       .value,
     githubWorkflowCheck := githubWorkflowCheck.dependsOn(ThisBuild / mergifyCheck).value,
-    ThisBuild / mergifyLabelPaths := {
-      val labelPaths = (ThisBuild / mergifyLabelPaths).value
+    internallyCalculatedMergifyLabelPaths := {
```
Comment (armanbilge) [inline mergify/src/main/scala/org/typelevel/sbt/mergify/MergifyPlugin.scala:118]: You still need the `ThisBuild` :)
```suggestion
    ThisBuild / internallyCalculatedMergifyLabelPaths := {
```

Comment (TonioGela) [reply mergify/src/main/scala/org/typelevel/sbt/mergify/MergifyPlugin.scala:118]: 😎

**Comment by bpholt** [comment]
> Hmm, running `mergifyGenerate` annihilates `.mergify.yml`, so the pr logic is probably wrong 👉 👈

I wonder if we ought to have some `scripted` tests for things like this, to show that the generated file is correct in different scenarios. (Not that you _have_ to add that to this PR if you don't want to, just that it might be helpful if you did.)

**Comment by TonioGela** [comment]
I don't have that much experience with scripted tests if not in g8 templates, but AFAIK, the series/0.5 has some tests I might take inspiration from, I'll check and add them

**Comment by armanbilge** [comment]
Yes, we _really_ ought to have tests in general. I've been lazy to do so, partly because testing generated output can become annoying. See some discussion in https://github.com/typelevel/sbt-typelevel/issues/26.

For this specific one, we can use a dog-food: override `mergifyLabelPaths` in `build.sbt`, just to prove we can. For example you should be able to remove this useless label.
- https://github.com/typelevel/sbt-typelevel/blob/abc217d7173ec51f44ed510e7d8cc5819543e7f2/.mergify.yml#L156-L163

**Review by armanbilge — APPROVED** [review]
Happy to close this one, finally! This was a good exercise in sbt setting evaluation and scoping 🤓

**Comment by TonioGela** [comment]
> For this specific one, we can use a dog-food: override `mergifyLabelPaths` in `build.sbt`, just to prove we can. For example you should be able to remove this useless label.

I did it and it might act as a test as you said. For what concerns #26, probably the better thing is to implement them in a separate PR, innit?

**Pull request merged**